Add docs for Snap (#555)
getsolus/help-center-docs: docs/user/software/third-party/snap.md

---
title: Snap
summary: Working with Snaps
---

# Snap
:::warning
Note that Snaps are going to be [removed from Solus in the future][2],

> and are not fully confined in Solus with kernels newer than 6.9.

Reviewer: 6.9 is the first kernel version where Snaps will not be confined; the warning makes it sound like they'll still be confined with 6.9.

Author: Whoops. Rephrased!

See the section on confinement below.
:::

[Snaps][1] are a way of installing third-party packages.
The `snapd` package installs the CLI client that can be used to install and run snaps.

## Confinement warning

> Snaps are usually protected using `strict` confinement,
> this ensures that Snaps can't access more of your system than needed.

Reviewer: ```suggestion
Snaps are usually protected using `strict` confinement,
ensuring that Snaps can't access more of your system than needed.
```

The `snap` command provided by Solus shows a warning when strict confinement is not available.
For example:

> WARNING: snap is running with partial confinement. See https://help.getsol.us/docs/user/software/third-party/snap for details

We recommend [migrating to Flatpak](#migrating-to-flatpak) when possible.
The warning can be silenced by running `sudo snap hide-confinement-warning`.

## Migrating to Flatpak

The `unsnap` package can be used to migrate from Snap to Flatpak.
While it does not actively remove user data in `~/snap` or `/var/lib/snapd`,
we cannot guarantee that no data gets removed on accident.

> Please ensure you have system backups that include snap data.

Reviewer: When referring to Snaps, is the S always capitalized? I guess the proper question is, is it a noun or a proper noun?

Author: Based on snapcraft.io it isn't. Updated!

Run `sudo unsnap` before migrating to see if all your snaps have known equivalents.
Look for lines containing `No equivalent flatpak for <name> found`.
Check out [this issue][3] if you encounter such snaps.

Running `sudo unsnap auto` will migrate as many Snaps to Flatpak as possible.
It will also uninstall `snapd` if all snaps were migrated.

> If `unsnap` is unable to find an equivalent Flatpak for a snap you have the following options:

Reviewer: ```suggestion
If `unsnap` is unable to find an equivalent Flatpak for a snap, you have the following options:
```

> - Search for an equivalent Flatpak manually and [report it][3] if possible.

Reviewer: ```suggestion
- Search for an equivalent Flatpak manually, and [report it][3] if possible.
```

---
title: Snap
summary: Working with Snaps
---

# Snap
:::warning
Note that Snaps are going to be [removed from Solus in the future][2],
and are not fully confined in Solus with kernels newer than 6.9.
See the section on confinement below.
:::

[Snaps][1] are a way of installing third-party packages.
The `snapd` package installs the CLI client that can be used to install and run snaps.

## Confinement warning

Snaps are usually protected using `strict` confinement,
this ensures that Snaps can't access more of your system than needed.
The `snap` command provided by Solus shows a warning when strict confinement is not available.
For example:

> WARNING: snap is running with partial confinement. See https://help.getsol.us/docs/user/software/third-party/snap for details

We recommend [migrating to Flatpak](#migrating-to-flatpak) when possible.
The warning can be silenced by running `sudo snap hide-confinement-warning`.

## Migrating to Flatpak

The `unsnap` package can be used to migrate from Snap to Flatpak.
While it does not actively remove user data in `~/snap` or `/var/lib/snapd`,
we cannot guarantee that no data gets removed on accident.

> Please ensure you have system backups that include snap data.

Reviewer 2: It would probably make sense to redo this so a combination of line 33 and 35 represents a single notice?

Author: Whoops, line 35 was my first draft of this sentence :see_no_evil:

This does not neccesarily preserve/migrate user data, so take care to make a backup beforehand.

Run `sudo unsnap` before migrating to see if all your snaps have known equivalents.
Look for lines containing `No equivalent flatpak for <name> found`.
Check out [this issue][3] if you encounter such snaps.

> Running `sudo unsnap auto` migrate as many Snaps to Flatpak as possible.

Reviewer 2: "will migrate" mights sound better here?
